### ISSUE TITLE
multiple reload of page cause intervention to be disruptive on favicon.ico

### ISSUE THREAD

**Issue by ampminfo** [SpiderLabs/ModSecurity#2976]
I was testing my app in Golang with ModSecurity 3.0.10.  I was refreshing the page multiple time to test something and my server start giving me a 500 internal error returned but my inspection from my modsecurity implementation. After some test I found in my modsecurity_debug.log the error that procude the intervention.  After 20+ reload, the crash occur on the following rule. I was only reloading the login page of the application. Nothing to block there

The questions are: 
1) Even after a reload of my executable of the app. The blocking still occurs and always related to that favicon.ico. Why and where are the memory of that blocking are stored.  
2) Why this rule trigger only after multiple reload and is kept blocked after a restart of the application.

[169308372743.594574] [/favicon.ico] [9]  T (0) t:urlDecodeUni: "GA1.1.849814932.1680637192"
[169308372743.594574] [/favicon.ico] [9] Target value: "GA1.1.849814932.1680637192" (Variable: REQUEST_COOKIES:_ga)
[169308372743.594574] [/favicon.ico] [9]  T (0) t:urlDecodeUni: "GS1.1.1691772677.22.1.1691773556.0.0.0"
[169308372743.594574] [/favicon.ico] [9] Target value: "GS1.1.1691772677.22.1.1691773556.0.0.0" (Variable: REQUEST_COOKIES:_ga_CZJH9BPMEF)
[169308372743.594574] [/favicon.ico] [9]  T (0) t:urlDecodeUni: "MTY5MzA4MjkwOXxOd3dBTkVJeVRWZFhWRkZKU1VaRVVWTlhTRlZTTmxOVVZGZEJSRWRNUXpWVFNVbzBU (90 characters omitted)"
[169308372743.594574] [/favicon.ico] [9] Target value: "MTY5MzA4MjkwOXxOd3dBTkVJeVRWZFhWRkZKU1VaRVVWTlhTRlZTTmxOVVZGZEJSRWRNUXpWVFNVbzBU (117 characters omitted)" (Variable: REQUEST_COOKIES:session)
[169308372743.594574] [/favicon.ico] [7] Added regex subexpression TX.0: -QSnDdXzHKPXy-i-
[169308372743.594574] [/favicon.ico] [7] Added regex subexpression TX.1: -QSnDdXzHKPXy-i-
[169308372743.594574] [/favicon.ico] [9] Matched vars updated.
[169308372743.594574] [/favicon.ico] [4] Running [independent] (non-disruptive) action: setvar
[169308372743.594574] [/favicon.ico] [8] Saving variable: TX:anomaly_score_pl4 with value: 0
[169308372743.594574] [/favicon.ico] [4] Running [independent] (non-disruptive) action: setvar
[169308372743.594574] [/favicon.ico] [8] Saving variable: TX:sql_injection_score with value: 0
[169308372743.594574] [/favicon.ico] [9]  T (0) t:urlDecodeUni: "_ga"
[169308372743.594574] [/favicon.ico] [9] Target value: "_ga" (Variable: REQUEST_COOKIES_NAMES:_ga)
[169308372743.594574] [/favicon.ico] [9]  T (0) t:urlDecodeUni: "_ga_CZJH9BPMEF"
[169308372743.594574] [/favicon.ico] [9] Target value: "_ga_CZJH9BPMEF" (Variable: REQUEST_COOKIES_NAMES:_ga_CZJH9BPMEF)
[169308372743.594574] [/favicon.ico] [9]  T (0) t:urlDecodeUni: "session"
[169308372743.594574] [/favicon.ico] [9] Target value: "session" (Variable: REQUEST_COOKIES_NAMES:session)
[169308372743.594574] [/favicon.ico] [4] Rule returned 1.
[169308372743.594574] [/favicon.ico] [9] Running action: log
[169308372743.594574] [/favicon.ico] [9] Saving transaction to logs
[169308372743.594574] [/favicon.ico] [9] Running action: auditlog
[169308372743.594574] [/favicon.ico] [9] Saving transaction to logs
[169308372743.594574] [/favicon.ico] [9] Running action: status
[169308372743.594574] [/favicon.ico] [4] Running (non-disruptive) action: tag
[169308372743.594574] [/favicon.ico] [9] Rule tag: application-multi
[169308372743.594574] [/favicon.ico] [4] Running (non-disruptive) action: tag
[169308372743.594574] [/favicon.ico] [9] Rule tag: language-multi
[169308372743.594574] [/favicon.ico] [4] Running (non-disruptive) action: tag
[169308372743.594574] [/favicon.ico] [9] Rule tag: platform-multi
[169308372743.594574] [/favicon.ico] [4] Running (non-disruptive) action: tag
[169308372743.594574] [/favicon.ico] [9] Rule tag: attack-sqli
[169308372743.594574] [/favicon.ico] [4] Running (non-disruptive) action: tag
[169308372743.594574] [/favicon.ico] [9] Rule tag: OWASP_CRS
[169308372743.594574] [/favicon.ico] [4] Running (non-disruptive) action: tag
[169308372743.594574] [/favicon.ico] [9] Rule tag: capec/1000/152/248/66
[169308372743.594574] [/favicon.ico] [4] Running (non-disruptive) action: tag
[169308372743.594574] [/favicon.ico] [9] Rule tag: PCI/6.5.2
[169308372743.594574] [/favicon.ico] [4] Running (non-disruptive) action: tag
[169308372743.594574] [/favicon.ico] [9] Rule tag: paranoia-level/4
[169308372743.594574] [/favicon.ico] [9] This rule severity is: 4 current transaction is: 255
[169308372743.594574] [/favicon.ico] [9] Saving msg: Restricted SQL Character Anomaly Detection (cookies): # of special characters exceeded (3)
[169308372743.594574] [/favicon.ico] [4] Running (disruptive)     action: block.
[169308372743.594574] [/favicon.ico] [8] Marking request as disruptive.
[169308372743.594574] [/favicon.ico] [8] Running action deny
[169308372743.594574] [/favicon.ico] [8] Skipping this phase as this request was already intercepted.

**Expected behavior**

A way to clear the memory within application or at least on every restart

**Server (please complete the following information):**
 - ModSecurity version (and connector): 3.0.10
 - WebServer: golang 1.20.7 with Echo Framework 4.11.1
 - OS (and distro): Wubuntu 11.3 last updates

**Rule Set (please complete the following information):**
odsecdefault.conf
crs-setup.conf
REQUEST-913-SCANNER-DETECTION.conf
REQUEST-942-APPLICATION-ATTACK-SQLI.conf

**Additional context**

This intervention seems also not to return any log output from the SetServerLogCallback function.
[modsecurity_debug.log](https://github.com/SpiderLabs/ModSecurity/files/12446734/modsecurity_debug.log)

**Comment by ampminfo** [comment]
crs-setup.conf modifiied lines

SecDefaultAction "phase:1,log,auditlog,deny,status:403"
SecDefaultAction "phase:2,log,auditlog,deny,status:403"

**Comment by martinhsv** [comment]
Hello @ampminfo ,

You may wish to inquire with the rule publisher about this rule, as they would know more about any issues with it that have been experienced by other users.  I'll note, in general, however:
- 942421 is a CRS rule with paranoia level 4; rules with paranoia levels of 3 or 4 are known to be highly prone to false positives
- it appears that the tested string is truncated, so it is entirely possible that there are 3+ 'forbidden' characters (as defined by the rule) in the string
- given that it is a cookie value that could explain why it only occurs on some requests but not all. Cookie values, by their nature vary, and may sometimes have a certain combination of characters but not other times
- the other thing that is notable about the rule is that there are characters in the regular expression that are not part of the ASCII set (0x00-0xF7). That is unusual and it might be possible that that is causing some anomalous effects.

**Comment by martinhsv** [comment]
With no further activity, I'll go ahead and close this.
